scripts/use_pyc.py

```diff
@@ -0,0 +1,81 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import logging
+import glob
+import os
+import platform
+import re
+import sys
+from pathlib import Path
+import shutil
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def calculate_folder_size(start_path):
+    """Calculate total size of a folder and file count."""
+    # https://stackoverflow.com/questions/1392413/calculating-a-directorys-size-using-python
+    total_size = 0
+    total_count = 0
+    for dirpath, dirnames, filenames in os.walk(start_path):
+        for f in filenames:
+            fp = os.path.join(dirpath, f)
+            # skip if it is symbolic link
+            if not os.path.islink(fp):
+                total_count += 1
+                total_size += os.path.getsize(fp)
+
+    return total_size, total_count
+
+
+def _print_folder_size(folder):
+    size, count = calculate_folder_size(folder)
+    size_in_mb = size / 1048576  # 1 MB = 1024 * 1024 B = 1048576 B
+    _LOGGER.info(f"{size_in_mb:.2f} MB, {count} files")
+
+
+def main(folder, version=None):
+    _LOGGER.info(f'Replace .py with .pyc, base folder: {folder}')
+    _print_folder_size(folder)
+    if version is None:
+        version = re.search(r'python(\d\.\d+)', folder).group(1)
+    else:
+        # 3.10.10
+        version = '.'.join(version.split('.')[:2])
+    # invoke==1.2.0 has a weird file: invoke/completion/__pycache__/__init__.cpython-36.pyc
+    # define pyc suffix to skip it
+    pyc_suffix = f'cpython-{version.replace(".", "")}.pyc'
+    _LOGGER.info(f'pyc suffix: {pyc_suffix}')
+    for file in glob.glob(f'{folder}/**/__pycache__/*{pyc_suffix}', recursive=True):
+        # If pip's py files are also removed, the error is raised when installing some packages.
+        # See https://github.com/Azure/azure-cli/pull/25801 for details.
+        if os.path.join('site-packages', 'pip') in file:
+            continue
```

Comment:
Removing py file has side effects.
For example: `/opt/az/bin/python3 -m pip install --no-cache  pgcli==3.4.1`
```
× pip subprocess to install build dependencies did not run successfully.
│ exit code: 2
╰─> [1 lines of output]
    /opt/az/bin/python3: can't open file '/opt/az/lib/python3.10/site-packages/pip/__pip-runner__.py': [Errno 2] No such file or directory
    [end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
```
The MSI also keeps py files under `pip` folder.